internal/validation/gateway/httproute.go

```diff
@@ -181,3 +187,55 @@ func getListenersForHTTPRouteValidation(sectionName *gatewayv1beta1.SectionName,
 
 	return listenersForValidation, nil
 }
+
+func validateWithKongGateway(
+	ctx context.Context, routesSvc kong.AbstractRouteService, parserFeatures parser.FeatureFlags, kongVersion semver.Version, httproute *gatewayv1beta1.HTTPRoute,
+) (bool, string, error) {
+	// Translate HTTPRoute to Kong Route object(s) that can be sent directly to the Admin API for validation.
+	// Use KIC parser that works both for traditional and expressions based routes.
+	var kongRoutes []kong.Route
+	var errMsgs []string
+	for _, rule := range httproute.Spec.Rules {
+		var (
+			routes []kongstate.Route
+			err    error
+		)
+		translation := translators.KongRouteTranslation{
+			Name:    "validation-attempt",
+			Matches: rule.Matches,
+			Filters: rule.Filters,
+		}
+		routes, err = parser.GenerateKongRouteFromTranslation(
+			httproute, translation, parserFeatures.RegexPathPrefix, parserFeatures.ExpressionRoutes, kongVersion,
+		)
+		if err != nil {
+			errMsgs = append(errMsgs, err.Error())
+			continue
+		}
+		for _, r := range routes {
+			kongRoutes = append(kongRoutes, r.Route)
+		}
+	}
+	if len(errMsgs) > 0 {
+		return false, validationMsg(errMsgs), nil
+	}
+	// Validate by using feature of Kong Gateway.
+	for _, kg := range kongRoutes {
+		kg := kg
+		ok, msg, err := routesSvc.Validate(ctx, &kg)
+		if err != nil {
+			return false, "unable to validate HTTPRoute schema", nil //nolint:nilerr
+		}
```

Comment:
- It might be beneficial to log the `err`. WDYT?
- What if we can a transient network error? Do we retry or fail validation hence rejecting the `HTTPRoute`?

Reply:
In terms of retries etc. for such cases I rely on go-kong implementation it's done the same way in others validators too. Maybe it's worth to adress in separate PR. Btw. I see area for small refactors in this part of our codebase to do as standalone task.

Instead of logging I think we can return it